### ISSUE TITLE
Only 1 boot entry

### ISSUE THREAD

**Issue by joihn** [rw-hsma-fpga/grub-switch#1]
thanks for the repo,
small issue on 
` grub-install (GRUB) 2.04-1ubuntu26.15` on ubuntu 20.04, 

only 1 boot entry listed by the tool :/ 

```
Switch positions chosen:
------------------------
0 : GRUB Menu (Fixed)
1 :
2 :
3 :
4 :
5 :
6 :
7 :
8 :
9 :
a :
b :
c :
d :
e :
f :
--------------------------------------------
* Press Enter to confirm selection
* Press Backspace <- to change configuration
* Press q to quit without changes
```

but my `grub customizer` can see many more:
![image](https://user-images.githubusercontent.com/11663917/208989302-b84e6ca7-bbd5-4fb7-97e9-9c5ceb2e0992.png)

I can post my  ```/boot/grub/grub.cfg```   here if needed (don't know if that file contains sensitive info?)

**Comment by joihn** [comment]
the entry the tool is not seeing (among other) is:
```
menuentry "Windows Boot Manager (on /dev/nvme0n1p2)" --class windows --class os $menuentry_id_option 'osprober-efi-403C-EC14' {
	insmod part_gpt
	insmod fat
	if [ x$feature_platform_search_hint = xy ]; then
	  search --no-floppy --fs-uuid --set=root  403C-EC14
	else
	  search --no-floppy --fs-uuid --set=root 403C-EC14
	fi
	chainloader /EFI/Microsoft/Boot/bootmgfw.efi
}
```

**Comment by joihn** [comment]
the entry the tool is seeing:
```
menuentry 'UEFI Firmware Settings' $menuentry_id_option 'uefi-firmware' {
	fwsetup
}
```

**Comment by rw-hsma-fpga** [comment]
Apologies, totally missed this issue post. Changed my notification settings now and hope that doesn't happen again.


Figuring out things by email with the original poster now - putting the solution here when we've figured it out.

**Comment by rw-hsma-fpga** [comment]
Appears to be fixed in most recent commit [e8f9b73](https://github.com/rw-hsma-fpga/grub-switch/commit/e8f9b737731c96dc34d488b824cdf55009a5e23b).

The regular expression in the **grub.cfg** parser was only looking for titles between apostrophes/ticks, but some versions of **os-prober** apparently produce them with quotes. RegEx is extended now.

Also apparently, there does not need to be a space between a submenu title and the opening bracket:

    submenu "Advanced Ubuntu options"{

so removed that condition from the RegEx too.
